### PR TITLE
o/configstate/configcore/vitality: fix RequireMountedSnapdSnap bug

### PR DESCRIPTION
**Pull request by anonymouse64**
There currently is a bug where if you apply vitality-hint to a snap service on
UC18+, the service unit will be overwritten without the Requires that prevents
other bugs around starting up very early before snapd is available.

The fix for this is to apply the same logic to configcore vitality handler,
so that the proper set of options is used when re-generating service units that
are included in the vitality-ranking.

Also get rid of an unnecessary loop which was rewriting the service units too
many times.

Finally, add a simple regression spread test about this scenario.

### CODE REVIEW

**Review by pedronis — DISMISSED** [review]
thanks

**Comment by pedronis** [comment]
marking blocked because it might interact with other fixes we are working on

**Review by mvo5 — DISMISSED** [review]
Thank you

**Comment by anonymouse64** [comment]
This needs to be updated now that https://github.com/snapcore/snapd/pull/10171 has landed

**Comment by anonymouse64** [comment]
This is ready for review again, some slight tweaks now that things have landed

**Review by pedronis — APPROVED** [review]
thanks, I fixed the spread test (Requires->Wants) and added the known TODO

**Review by mvo5 — APPROVED** [review]
Thank you

File: overlord/configstate/configcore/vitality.go [hunk 1]
```diff
@@ -58,6 +58,8 @@ func handleVitalityConfiguration(tr config.Conf, opts *fsOnlyContext) error {
 	st.Lock()
 	defer st.Unlock()
 
+	// TODO: Reimplement most of this as a servicestate.UpdateVitalityRank
```
Comment (anonymouse64) [inline overlord/configstate/configcore/vitality.go:61]: :+1:

**Pull request merged**